src/shared/schema/schemas.ts

```diff
@@ -0,0 +1,130 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdirSync, writeFileSync } from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as filesystemUtilities from '../filesystemUtilities'
+import { getLogger } from '../logger'
+import { CompositeResourceFetcher } from '../resourcefetcher/compositeResourceFetcher'
+import { FileResourceFetcher } from '../resourcefetcher/fileResourceFetcher'
+import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
+import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { normalizeSeparator } from '../utilities/pathUtils'
+import { Schemas } from './schemaService'
+
+const GOFORMATION_MANIFEST_URL = 'https://api.github.com/repos/awslabs/goformation/releases/latest'
+
+export type SchemaType = 'cfn' | 'sam' | 'none'
+
+/**
+ * Loads JSON schemas for CFN and SAM templates.
+ * Checks manifest and downloads new schemas if the manifest version has been bumped.
+ * Uses local, predownloaded version if up-to-date or network call fails
+ * If the user has not previously used the toolkit and cannot pull the manifest, does not provide template autocomplete.
+ * @param extensionContext VSCode extension context
+ */
+export async function getSchemas(extensionContext: vscode.ExtensionContext): Promise<Schemas> {
+    try {
+        // Convert the paths to URIs which is what the YAML extension expects
+        const cfnSchemaUri = vscode.Uri.file(
+            normalizeSeparator(path.join(extensionContext.globalStoragePath, 'cloudformation.schema.json'))
+        )
+        const samSchemaUri = vscode.Uri.file(
+            normalizeSeparator(path.join(extensionContext.globalStoragePath, 'sam.schema.json'))
+        )
+        const goformationSchemaVersion = await getTag(GOFORMATION_MANIFEST_URL)
+
+        await getRemoteOrCachedFile({
+            filepath: cfnSchemaUri.fsPath,
+            version: goformationSchemaVersion,
+            url: `https://raw.githubusercontent.com/awslabs/goformation/${goformationSchemaVersion}/schema/cloudformation.schema.json`,
+            cacheKey: 'cfnSchemaVersion',
+            extensionContext,
+        })
+        await getRemoteOrCachedFile({
+            filepath: samSchemaUri.fsPath,
+            version: goformationSchemaVersion,
+            url: `https://raw.githubusercontent.com/awslabs/goformation/${goformationSchemaVersion}/schema/sam.schema.json`,
+            cacheKey: 'samSchemaVersion',
+            extensionContext,
+        })
+        return {
+            cfn: cfnSchemaUri,
+            sam: samSchemaUri,
+        }
+    } catch (e) {
+        getLogger().error('Could not refresh schemas:', (e as Error).message)
+        throw e
+    }
+}
+
+/**
+ * Retrieves tag name from a GitHub release manifest
+ * @param manifestUrl release manifest url
+ * @returns tag name
+ */
+export async function getTag(manifestUrl: string, fetcher?: HttpResourceFetcher): Promise<string | undefined> {
```

Comment:
consider moving this to shared/utilities/github.ts (new module). It's not specific to CFN json schemas. 

OTOH, this has a lot of opportunity for generalization. Most of the code is dealing with ResourceFetcher's boilerplate, the rest JSON navigation. I could imagine a util function like:

    getFromJsonUri(manifestUrl, 'tag_name')

Which does the business of downloading any random JSON file and getting something from it. Could later be extended to take queries. But by then we would be in library-land.